components/eam/src/physics/cam/shoc.F90

```diff
@@ -3054,7 +3054,7 @@ subroutine adv_sgs_tke(nlev, shcol, dtime, shoc_mix, wthv_sec, &
         a_prod_sh=tk(i,k)*sterm_zt(i,k)
 
         ! Dissipation term
-        a_diss(i,k)=Cee/shoc_mix(i,k)*tke(i,k)**1.5
+        a_diss(i,k)=Cee/shoc_mix(i,k)*bfb_pow(tke(i,k),1.5_rtype)
```

Comment:
Here I recommend `bfb_sqrt(bfb_cube(tke(i,k)))`, but I've never actually demonstrated it should be preferred, so it's just a suggestion.

Reply:
I can make this change if it is better than using `pow`. For C++ code, I think we have `ekat::sqrt` and `ekat::cube` which we can use to mimic this calculation.

Reply:
I did some investigating and have firmer evidence that `bfb_sqrt(bfb_cube(tke(i,k)))` is the way to go:

- I wrote a short program and looked at the assembly with `-xcore-avx512 -O3 -fp-model $model -fsource-asm -S -c`, where `$model` is either `fast` or `strict`. In the `fast` case, ifort on Compy ends up emitting code for the sqrt(x^3) formulation instead of pow, supporting that it's faster.
- Then I ran a bunch of trials with `strict` on random numbers drawn from U[0,1] and found that it's also slightly more accurate. I did this by computing these expressions in single precision and comparing with a double-precision truth value. But only slightly: The pow formulation is worse 10.3% of the time, the sqrt 7.2%, and it's a tie the rest. But slightly better and faster makes sqrt(x^3) the way to go.

Reply:
My apologies for being quick on the merge..